samples/start-app/php-symfony/src/Controller/AppActivationCallbackController.php

```diff
@@ -16,23 +19,75 @@ final class AppActivationCallbackController extends AbstractController
     public function __construct(
         private readonly AppActivationCallback    $callback,
         private readonly TokenRepositoryInterface $tokenRepository,
-        private readonly string                   $projectDir)
+        private readonly UserRepositoryInterface  $userRepository,
+        private readonly LoggerInterface          $logger,
+        private readonly string                   $projectDir,
+        private readonly string                   $openIdAuthentication)
     {
     }
 
     #[Route('/callback', name: 'callback', methods: ['GET'])]
     public function __invoke(): Response
     {
         session_start();
-        $this->callback->execute($_SESSION, $_GET['state'], $_GET['code']);
 
-        if (!$this->tokenRepository->hasToken()) {
-            return new Response(file_get_contents($this->projectDir . '/templates/no_access_token.html'));
-        }
+        try{
+            $data = $this->callback->execute($_SESSION, $_GET['state'], $_GET['code']);
+
+            $response = new Response();
+
+            $divToReplace = "<div>UserInformation</div>";
+            $divUserInformation = "";
+
+            if (!$this->tokenRepository->hasToken()) {
+                return $response->setContent(
+                    str_replace(
+                        $divToReplace,
+                        $divUserInformation,
+                        file_get_contents($this->projectDir . '/templates/no_access_token.html')
+                    )
+                );
+            }
+
+            if ($this->openIdAuthentication && isset($data['id_token'])) {
+                $response = $this->forward('App\Controller\OpenIdController::extractUserInformation', [
```

Comment:
I think you can directly call the usecase here and import the cookie creation just below

Reply:
Done